### PR TITLE
Movement implementation

### PR DESCRIPTION
**Pull request by liran-funaro**
Signed-off-by: Liran Funaro <liran.funaro@gmail.com>

### CODE REVIEW

File: internal/tokens/fungible.go [hunk 1]
```diff
@@ -234,3 +308,81 @@ func (ctx *FungibleTxContext) transfer(
 
 	return newRecord, nil
 }
+
+func (ctx *FungibleTxContext) ltx() (bcdb.Ledger, error) {
```
Comment (tock-ibm) [inline internal/tokens/fungible.go:312]: took me a minute...
`ledgerTx()` is a better name

File: internal/tokens/manager.go [hunk 2]
```diff
@@ -1469,7 +1470,146 @@ func (m *Manager) FungibleAccounts(typeId string, owner string, account string)
...
+	mainAccountKey := getAccountKey(owner, mainAccount)
+	allHistory, err := ptx.GetHistoricalData(ctx.dbName, mainAccountKey)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read account's historical data")
+	}
+
+	// We sort the values from newest to oldest (the highest version is first)
+	sort.SliceStable(allHistory, func(i, j int) bool {
+		return versionCompare(allHistory[i].Metadata.Version, allHistory[j].Metadata.Version) > 0
+	})
```
Comment (tock-ibm) [inline internal/tokens/manager.go:1500]: Is it really unordered? if it is ordered old-new we could just iterate from the end

Comment (liran-funaro) [reply internal/tokens/manager.go:1500]: In the test, I witness it was ordered from old to new, but I did not find any indication of that in the documentation or in the code. So I prefer not to rely on that.

File: internal/tokens/manager.go [hunk 3]
```diff
@@ -1469,7 +1470,146 @@ func (m *Manager) FungibleAccounts(typeId string, owner string, account string)
...
+		if err != nil {
+			return nil, err
+		}
+
+		if op.ownerRead[mainAccountKey] == nil || op.ownerWrite[mainAccountKey] == nil {
+			return nil, common.NewErrInternal("movement TX referred to a data tx that do not update the main account: %+v", op)
+		}
+
+		if len(op.otherWrite) > 1 {
+			return nil, common.NewErrInternal("movement TX updates more than one account")
```
[the remark this answers is not in the document]
Comment (liran-funaro) [reply internal/tokens/manager.go:1540]: OK. I'll change it to `DestinationAccounts`. In the tests, I'll make sure it is just one account (for now).

File: internal/tokens/fungible.go (hunk 1 above)
Comment (liran-funaro) [reply internal/tokens/fungible.go:312]: OK

File: internal/tokens/manager.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (liran-funaro) [reply internal/tokens/manager.go]: I changed the movement to include a list of destination and/or source accounts.
So now the action type and value are implicit. The value is the sum of all sources/destinations' quantity.

File: internal/tokens/manager.go (hunk 2 above)
Comment (tock-ibm) [reply internal/tokens/manager.go:1500]: ok

**Review by tock-ibm — APPROVED** [review]

**Pull request merged**